app/migration/ecf2_teacher_history.rb

```diff
@@ -18,43 +18,104 @@ def initialize(teacher_row:, ect_at_school_period_rows: [], mentor_at_school_per
     @mentor_at_school_period_rows = mentor_at_school_period_rows
   end
 
+  def save_all!
+    teacher = find_or_create_teacher!
+    save_ect_periods!(teacher)
+    save_mentor_periods!(teacher)
+    teacher
+  end
```

Comment:
This is an interesting one. I don't know if we'd want to do both at once during a real migration, because I think we need all the mentor ones to be done first, then the ECT ones can follow. If we don't do it that way round we won't be able to create `MentorshipPeriod` records unless the associated teacher has already been processed.